rfcs/0013-synthea-future-claims-and-automation.md

```diff
@@ -0,0 +1,130 @@
+
+Additionally, for this data to be ingested on a weekly basis, the BFD ingestion pipeline will need to work with Synthea data as it does with CCW data, for which back-dated beneficiary filtering will be turned off for Synthea data. 
+
+When producing and loading a large set of synthetic data and performance testing:
+    - Size and shape of the data will reflect production
+    - Will be available for on-demand performance and load testing
+    - Will not be released. Specifically, not released to PROD.
+    - Will be isolated from other released synthetic data. 
+    - Will have the capability to be re-generated with future versions of synthea
+    - Synthetic beneficiaries for load and performance testing will not overlap with any beneficiaries used by end users
+
+For Part D contract IDs, the value will not need to change on a regular basis unless there is a specific customer use case presented.
+
+### Proposed Solution: Detailed Design
+[Proposed Solution: Detailed Design]: #proposed-solution-detailed-design
+
+Weekly Generation Plan:
+
+* On a weekly basis a CRON job within Jenkins will be set up to run scripts that:
+  - Clone latest version of Mitre Synthea code
+  - Run currently manual steps in the `Synthea Test Plan` such as accessing and comparing the latest Synthea end-properties file, running queries in PROD SBX to determine database ranges for beneficiary and claim properities, and ensuring the next batch of data will not overlap with existing data ranges.
+  - Generate a batch of synthetic data and load into AWS S3 to trigger the BFD pipeline to transform and load synthetic data into database for TEST, PROD SBX, and PROD.
+
+* A new database table 'synthetic-future-claims' will be created for staging future claims in TEST to be loaded at a later time:
+
+* Synthea data will be compatible with the ETL ingestion pipeline by modifying the manifest to inform that a load with back-dated data should be loaded.
+
+Load and Performance Testing Plan:
+- Resources: The BFD TEST and/or PROD SBX servers will be more utilized for on-demand performance and load testing
+
+- Scalability: Size and shape of the data will reflect production i.e. 1 million+ beneficiaries. This could take several days, however from previous load testing, it is possible to extrapolate how much time different load sizes will approximately take. 
+
+- Availability: Data will not be released to PROD without further approval. New data will be isolated from previously released synthetic data by a dynamic range of benficiary IDs, which also prevents overlap. 
+
+- Repeatability: Several delete SQL queries will be executed after load tests are completed, which will allow for beneficiary and claim data to be re-generated with future versions of synthea
+
+### Proposed Solution: Unresolved Questions
+[Proposed Solution: Unresolved Questions]: #proposed-solution-unresolved-questions
+
+What exactly will vary between weekly and quarterly communications to partners?
```

Comment:
Quarterly release will likely include changes to schema, fields, field values - and weekly releases will be announced in advance via a standard cadence message, e.g. "Weekly synthetic releases will occur on ___day each week before/after ___ o'clock (timezone)." and will only be updated when a specific weekly release is moved or cancelled.